modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java

```diff
@@ -441,33 +443,28 @@ private String getServerPID() throws BallerinaTestException {
             }
             tmp.destroy();
         } else {
-            BufferedReader bufferedReader = null;
-            FileReader fileReader = null;
-
+            //reading the process id from netstat
+            Process tmp;
             try {
-                //reading the pid form  carbon.pid file in server home dir
-                fileReader = new FileReader(serverHome + File.separator + Constant.SERVER_PID_FILE_NAME);
-                bufferedReader = new BufferedReader(fileReader);
-                pid = bufferedReader.readLine();
+                tmp = Runtime.getRuntime().exec("lsof -Pi tcp:" + httpServerPort);
             } catch (IOException e) {
-                throw new BallerinaTestException("Error retrieving pid from " + Constant.SERVER_PID_FILE_NAME);
-            } finally {
-                if (fileReader != null) {
-                    try {
-                        fileReader.close();
-                    } catch (IOException e) {
-                        //ignore
-                    }
+                throw new BallerinaTestException("Error retrieving netstat data", e);
+            }
+
+            String outPut = readProcessInputStream(tmp.getInputStream());
+            String[] lines = outPut.split(System.lineSeparator());
+            for (String line : lines) {
+                String[] column = line.trim().split("\\s+");
+                if (column.length < 10) {
+                    continue;
```

Comment:
checked that, I think we can't even use that, the reason is, it too gives both listening and connected processes PIDs and we cannot distinguish which is which( there may be other best approaches out there to do this directly, but didn't came across anything yet)